lib/shape.ml

```diff
@@ -58,6 +58,14 @@ let translate dx dy shape =
   | Ellipse ellipse -> Ellipse { ellipse with c = { x = ellipse.c.x + dx; y = ellipse.c.y + dy } }
   | Line line -> Line {a = {x = line.a.x + dx; y = line.a.y + dy}; b = {x = line.b.x + dx; y = line.b.y + dy}}
 
+let scale (factor:float) s =
+  let scaled_length len fact = int_of_float ((float_of_int len) *. sqrt(fact) +. 0.5) in
+  match s with
+  | Circle circle' -> circle ~x:circle'.c.x ~y:circle'.c.y (scaled_length circle'.radius factor)
+  | Rectangle rectangle' -> rectangle ~x:rectangle'.c.x ~y:rectangle'.c.y (scaled_length rectangle'.length factor) (scaled_length rectangle'.width factor)
+  | Ellipse ellipse' -> ellipse ~x:ellipse'.c.x ~y:ellipse'.c.y (scaled_length ellipse'.rx factor) (scaled_length ellipse'.ry factor)
+  | Line line' -> line ~x1:line'.a.x ~y1:line'.a.y (scaled_length line'.b.x factor) (scaled_length line'.b.y factor)
```

Comment:
This calculation is not correct for lines, because we don't have the length.

If you have figured out the calculation then go ahead and add it. I feel it'd be too much scope for one task, so feel free to create an issue for it.

For now, you can handle line by using `failwith` (https://ocaml.org/docs/error-handling#assertions)

e.g. `failwith "Not Implemented"`

Reply:
Yeah, I had doubts about Line. Thank you I will create an issue on this and we can discuss it separately. The rest will be done by tomorrow :)